lib/puppet/provider/cisco_bgp/nxapi.rb

```diff
@@ -199,6 +199,11 @@ def timer_bestpath_limit_set
     @bgp_vrf.timer_bestpath_limit_set(limit, always)
   end
 
+  def confederation_peers=(should_list)
+    should_list = @bgp_vrf.default_confederation_peers if should_list[0] == :default
+    @property_flush[:confederation_peers] = should_list
```

Comment:
It's baffling to me that you don't need `should_list.flatten` but I guess this is okay...